module2/lessons/intro_to_mvc.md

```diff
@@ -0,0 +1,134 @@
+    - `<% %>` does not render the return value. Use this when you don't want the return value to become part of the HTML. Typically, these are used for Ruby statements that control other statements.
+        - ex: `<% tasks.each do |task| %>` ... `<% end %>`
+        - ex: `<% if @tasks %>` ... `<% end %>`
+        
+
+## Putting it All Together
+
+- Someone sends a request to your application. That request includes:
+    - An HTTP verb (GET, POST, etc.)
+    - A URI path (`/tasks`, `/`, `/tasks/2`)
+- Rails inspects the verb/path combo and determines what controller action to call based on what routes you've defined in `routes.rb`
+- Within that controller action, we:
+    1. Perform any data manipulation we need using our model
+    2. Collect any data we need to use in our view (using a model as a go-between)
+    3. Render a view or redirect
+
+Let's draw a diagram to represent this process.
+
+<p>
+  <center>
+    <img src="../../assets/images/lessons/intro_to_mvc/mvc_rails.png" alt="Rails MVC Diagram" />
+  </center>
+</p>
+
+### Logic "Responsibilities"
+
+The MVC Design Pattern describes the primary responsibility of each portion of code that we'll describe a little bit deeper here.
+
+### Models -- Data Logic
+
+We saw earlier that the Model is responsible for interaction with the database. In MVC, we say that the Model is responsible for "data logic". *Calculations, filtering of data, or other manipulation of data should happen at the Model level.* When fetching data from a Model, the Model should only return raw data in an appropriate data structure -- usually an array, but can also be a hash.
+
+A Model generally does not alter data. For example, it would be appropriate for the Model to calculate the average age of all students, but it should NOT 'round' that data to, say, two decimal places. It can force the result to be a floating point number with `.to_f`, but it should return raw data as much as possible.
+
+### Views -- Presentation Logic
+
+Views have very little logic in them, generally just if/else statements and perhaps doing some basic iteration over a dataset. *The primary goal of a View is to manipulate raw data given to it by a Controller to present that data* in a way that is useful to our users.
+
+If a View has access to an instance variable, or a collection of instances in an array that it is iterating over, it's appropriate for the View to call "instance methods" from the Model class if needed. A View should not call *class* methods except in extremely rare cases where a data builder requires it, such as a drop-down select in a form for example.
+
+<section class="call-to-action">
+### MVC and Forms
+
+In MVC applications, we can get and manipulate data on an html page in multiple ways. You've collected data from a user with JavaScript eventListeners as part of the Hang In There project, and now we are using erb to get data from our view to our controller, as well as from our controller to our view.
```

Comment:
I like this add that references the Javascript event listeners that were introduced at the start of the mod!